libs/parallelism/algorithms/tests/unit/container_algorithms/unique_range.cpp

```diff
@@ -66,6 +68,66 @@ struct random_fill
 };
 
 ////////////////////////////////////////////////////////////////////////////
+void test_unique_sent()
+{
+    std::size_t const size = 10007;
+    std::vector<std::size_t> c(size), d;
+    std::generate(std::begin(c), std::end(c),
+        []() -> std::size_t { return std::rand() % 10; });
+    d = c;
+
+    auto end_len = std::rand() % 10006 + 1;
+    c[end_len] = 10;
+
+    auto result = hpx::ranges::unique(std::begin(c), sentinel<std::size_t>{10});
+    auto solution = std::unique(std::begin(d), std::begin(d) + end_len);
+
+    bool equality =
+        test::equal(std::begin(c), result.begin(), std::begin(d), solution);
+
+    HPX_TEST(equality);
+}
+
+template <typename ExPolicy>
+void test_unique_sent(ExPolicy policy)
+{
+    std::size_t const size = 10007;
+    std::vector<std::size_t> c(size), d;
+    std::generate(std::begin(c), std::end(c),
+        []() -> std::size_t { return std::rand() % 10; });
+    d = c;
+
+    auto end_len = std::rand() % 10006 + 1;
+    c[end_len] = 10;
+
+    auto result =
+        hpx::ranges::unique(policy, std::begin(c), sentinel<std::size_t>{10});
+    auto solution = std::unique(std::begin(d), std::begin(d) + end_len);
```

Comment:
Out of curiosity, why not `std::end(d)` as second parameter for `std::unique`?

Reply:
Well, the sentinel value could be anywhere right? So just wanted to test that.